### PR TITLE
Make announcing netplay on the public lobby optional

### PR DESCRIPTION
**Pull request by GregorR**
Like it says in the title. Adds a new configuration option to be private.

### CODE REVIEW

**Comment by fr500** [comment]
derp, wrong button :p

**Comment by Alcaro** [comment]
> \+      reannounce ++;

That's a ... unorthodox coding style.

**Comment by GregorR** [comment]
That's not my code, Alcaro, it just shows as an add because I moved that code line and reindented it. Don't blame me! ;)

**Comment by RobLoach** [comment]
#4552

File: network/netplay/netplay_frontend.c [hunk 1]
```diff
@@ -652,11 +652,22 @@ static void netplay_frontend_paused(netplay_t *netplay, bool paused)
...
-   reannounce ++;
-   if (netplay->is_server && (reannounce % 3600 == 0))
-      netplay_announce();
+   settings_t *settings  = config_get_ptr();
+
    retro_assert(netplay);
 
+   if (settings->netplay.public_announce)
+   {
+      reannounce ++;
```
Comment (RobLoach) [inline network/netplay/netplay_frontend.c]: It works though!

Comment (Alcaro) [inline network/netplay/netplay_frontend.c]: Yes, and so does [banks.c](http://www.ioccc.org/1998/banks.c).

Comment (RobLoach) [inline network/netplay/netplay_frontend.c]: I like planes.

**Pull request merged**